ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiLayeredWorldState.java

```diff
@@ -42,6 +45,8 @@
   protected final long height;
   protected final TrieLogLayer trieLog;
   private final Hash worldStateRootHash;
+  private final CodeCache codeCache =
+      new CodeCache(ContractCacheConfiguration.getInstance().getContractCacheWeight());
```

Comment:
I think this will be a problem.  A new layered worldstate gets created on every call to `BonsaiWorldStateArchive.persist`. I think we are going to end up with a lot of heap pressure from new codecache instances.  A singleton codecache should be safe and mitigate this.  @matkt, thoughts ?

Reply:
the number of layers in memory is limited to 512. So at most we will have 512 in memory. I don't know if the CodeCache is a large object. But maybe for this object we should create it only when we really use it. In many cases I think that the cache will be created in a layer but we do not need it or I miss something

Reply:
this is out of date, it has been replaced with a singleton for the cache.